### PR TITLE
feat: add Jira integration

### PR DESCRIPTION
**Pull request by burntcarrot**
* add basic HTTP client for Jira

Signed-off-by: burntcarrot <aadhav.n1@gmail.com>

### CODE REVIEW

File: integrations/jira/jira.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	req.Header.Set("Content-Type", "application/json")
+
+	// perform request
+	resp, err := client.Client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	// read response body
+	respBody, _ := io.ReadAll(resp.Body)
```
Comment (souvik-deepsource) [inline integrations/jira/jira.go:87]: Handle the error?

File: integrations/jira/jira.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	var parsed map[string]interface{}
+	json.Unmarshal([]byte(string(respBody)), &parsed)
+
+	// set issue key
+	issue.Key = parsed["key"].(string)
```
Comment (souvik-deepsource) [inline integrations/jira/jira.go:92]: What is JIRA's key format? Can you share?

**Review by souvik-deepsource — CHANGES_REQUESTED** [review]

File: integrations/jira/jira.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+	if resp.StatusCode == 201 {
+		return issue.Key, nil
+	} else {
+		return "", errors.New("Error:" + resp.Status)
+	}
```
Comment (souvik-deepsource) [inline integrations/jira/jira.go:98]:
```suggestion
	} 
	return "", errors.New("Error:" + resp.Status)
```

File: integrations/jira/jira.go (hunk 1 above)
Comment (burntcarrot) [reply integrations/jira/jira.go:87]: Oops! Nice catch, will fix :)

File: integrations/jira/jira.go (hunk 2 above)
Comment (burntcarrot) [reply integrations/jira/jira.go:92]: Sure! Here's an example of a [response](https://developer.atlassian.com/server/jira/platform/jira-rest-api-examples/#response).

**Comment by souvik-deepsource** [comment]
There are a couple of DeepSource issues that needs fixing.

**Comment by burntcarrot** [comment]
> There are a couple of DeepSource issues that needs fixing.

Thanks for the heads-up! Seems like my editor messed up :/

**Review by souvik-deepsource — APPROVED** [review]
Keep 'em coming! 🚀

**Pull request merged**